easybuild/easyblocks/t/tensorflow.py

```diff
@@ -0,0 +1,264 @@
+##
+# Copyright 2009-2017 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing TensorFlow, implemented as an easyblock
+"""
+import glob
+import os
+import stat
+import tempfile
+
+import easybuild.tools.environment as env
+import easybuild.tools.toolchain as toolchain
+from easybuild.easyblocks.generic.pythonpackage import PythonPackage
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, apply_regex_substitutions, mkdir, resolve_path
+from easybuild.tools.filetools import which, write_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+
+
+INTEL_COMPILER_WRAPPER = """#!/bin/bash
+export INTEL_LICENSE_FILE='%(intel_license_file)s'
+%(compiler_path)s "$@"
+"""
+
+
+class EB_TensorFlow(PythonPackage):
+    """Support for building/installing TensorFlow."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            # see https://developer.nvidia.com/cuda-gpus
+            'cuda_compute_capabilities': [[], "List of CUDA compute capabilities to build with", CUSTOM],
+            'with_mkl_dnn': [True, "Make TensorFlow use Intel MKL-DNN", CUSTOM],
```

Comment:
`mkl-dnn` and `imkl` are two different things. `mkl-dnn` is not part of the toolchain, and is totally independent from anything else (compiler, MPI, other math libraries). It can be automatically downloaded by Tensorflow during installation, which is what this option does.

Reply:
Yes, but you don't want it unless you are using mkl and especially not if you are using Cuda and cudnn.

Reply:
I am not sure what you mean. For sure you don't want it when using CUDA and cuDNN, but why wouldn't you want to use mkl-dnn without the full mkl?

Reply:
Misread the instructions a bit. No, you don't need mkl, but the instructions from TensorFlow and mkl-ddn is a bit unclear.

So, at the very least it should be turned off when CUDA and cuDNN are enabled.

Reply:
Agreed.